### PR TITLE
Wait & retry on rate limit errors for GitHub requests

### LINKED ISSUE #125: one more spot to wait/retry?
I guess I associated too many "things" with my tokens so I did manage to run out of free github API goodness:
```
  "resources": {                                                                                                                                                                                                     
    "core": {                                                                                                                                                                                                        
      "limit": 5000,                                                                                                                                                                                                 
      "used": 5000,                                                                                                                                                                                                  
      "remaining": 0,                                                                                                                                                                                                
      "reset": 1625262303                                                                                                                                                                                            
    },        
...
  "rate": {                                                                                                                                                                                                          
    "limit": 5000,                                                                                                                                                                                                   
    "used": 5000,                                                                                                                                                                                                    
    "remaining": 0,                                                                                                                                                                                                  
    "reset": 1625262303                                                                                                                                                                                              
  }    
```
which lead to
```
2021-07-02T17:40:02-0400 [INFO    ] tinuous Fetching runs newer than 2021-07-01 05:43:07+00:00                                                                                                                       
Traceback (most recent call last):                                                                                                                                                                                   
  File "/home/datalad/miniconda3/envs/tinuous-dev/bin/tinuous", line 33, in <module>                                                                                                                                 
    sys.exit(load_entry_point('tinuous', 'console_scripts', 'tinuous')())                                                                                                                                            
  File "/home/datalad/miniconda3/envs/tinuous-dev/lib/python3.9/site-packages/click/core.py", line 829, in __call__                                                                                                  
    return self.main(*args, **kwargs)                                                                                                                                                                                
  File "/home/datalad/miniconda3/envs/tinuous-dev/lib/python3.9/site-packages/click/core.py", line 782, in main                                                                                                      
    rv = self.invoke(ctx)                                                                                                                                                                                            
  File "/home/datalad/miniconda3/envs/tinuous-dev/lib/python3.9/site-packages/click/core.py", line 1259, in invoke                                                                                                   
    return _process_result(sub_ctx.command.invoke(sub_ctx))                                                                                                                                                          
  File "/home/datalad/miniconda3/envs/tinuous-dev/lib/python3.9/site-packages/click/core.py", line 1066, in invoke                                                                                                   
    return ctx.invoke(self.callback, **ctx.params)                                                                                                                                                                   
  File "/home/datalad/miniconda3/envs/tinuous-dev/lib/python3.9/site-packages/click/core.py", line 610, in invoke                                                                                                    
    return callback(*args, **kwargs)                                                                                                                                                                                 
  File "/home/datalad/miniconda3/envs/tinuous-dev/lib/python3.9/site-packages/click/decorators.py", line 33, in new_func                                                                                             
    return f(get_current_context().obj, *args, **kwargs)                                                                                                                                                             
  File "/mnt/datasets/datalad/ci/tinuous/src/tinuous/__main__.py", line 111, in fetch                                                                                                                                
    for obj in ci.get_build_assets(                                                                                                                                                                                  
  File "/mnt/datasets/datalad/ci/tinuous/src/tinuous/github.py", line 80, in get_build_assets                                                                                                                        
    for wf in self.get_workflows():                                                                                                                                                                                  
  File "/mnt/datasets/datalad/ci/tinuous/src/tinuous/github.py", line 67, in get_workflows                                                                                                                           
    for wf in self.ghrepo.get_workflows():                                                                                                                                                                           
  File "/home/datalad/miniconda3/envs/tinuous-dev/lib/python3.9/functools.py", line 969, in __get__                                                                                                                  
    val = self.func(instance)                                                                                                                                                                                        
  File "/mnt/datasets/datalad/ci/tinuous/src/tinuous/github.py", line 64, in ghrepo                                                                                                                                  
    return self.client.get_repo(self.repo)                                                                                                                                                                           
  File "/home/datalad/miniconda3/envs/tinuous-dev/lib/python3.9/site-packages/github/MainClass.py", line 330, in get_repo                                                                                            
    headers, data = self.__requester.requestJsonAndCheck("GET", url)                                                                                                                                                 
  File "/home/datalad/miniconda3/envs/tinuous-dev/lib/python3.9/site-packages/github/Requester.py", line 353, in requestJsonAndCheck                                                                                 
    return self.__check(                                                                                                                                                                                             
  File "/home/datalad/miniconda3/envs/tinuous-dev/lib/python3.9/site-packages/github/Requester.py", line 378, in __check                                                                                             
    raise self.__createException(status, responseHeaders, output)                                                                                                                                                    
github.GithubException.RateLimitExceededException: 403 {"message": "API rate limit exceeded for user ID 39889.", "documentation_url":                                                                                
"https://docs.github.com/rest/overview/resources-in-the-rest-api#rate-limiting"}      
```
while 
```
chronic flock -n -E 0 /home/datalad/.run/tinuous-datalad.lock /mnt/datasets/datalad/ci/logs/tools/cron_job 
```

I guess there might be one more spot to guard for `RateLimitExceededException` and have a nice nap?

### PR DESCRIPTION
**Pull request by jwodder**
Closes #125.

### CODE REVIEW

**Comment by jwodder** [comment]
The test failure is because the version of git-annex in Ubuntu's repositories is too old for DataLad.  #127 fixes that.

File: src/tinuous/base.py [hunk 1]
```diff
@@ -64,27 +64,41 @@ def from_travis_event(cls, travis_event: str) -> Optional["EventType"]:
...
+        while True:
             r = self.session.get(url, **kwargs)
-        r.raise_for_status()
-        return r
+            if r.status_code >= 500 and i < self.MAX_RETRIES:
+                log.warning(
+                    "Request to %s returned %d; waiting & retrying", url, r.status_code
+                )
+                i += 1
+                sleep(i)
```
Comment (yarikoptic) [inline src/tinuous/base.py:86]: This week result in up to about 50 sec sleep total with 10 retries. May be square it?

Comment (jwodder) [reply src/tinuous/base.py]: Done.

Comment (yarikoptic) [reply src/tinuous/base.py]: thanks!

**Comment by yarikoptic** [comment]
since no other pending PRs, let's just release.  Thank you @jwodder!

**Pull request merged**